### DISCUSSION TITLE
get env variables in config.yaml

### DISCUSSION THREAD

**Discussion by AnkitaBarbi** [dexidp/dex#2022] [Q&A] [answered]
I am using docker image ```ghcr.io/dexidp/dex:v2.27.0```
and I have config.yaml file where I want connectors and storage (postgres) to be fetched from env variables. How do I achieve it?

**Comment by nabokihms** [comment, marked as answer]
I believe that you can not do it with v2.27.0.
We tried to solve this problem in the upcoming release by adding `gomplate` into the docker image.

You need two things to make it work right now:
1. use ghcr.io/dexidp/dex:latest
2. add environment variables to your config like in the example here:
https://github.com/dexidp/dex/blob/master/config.docker.yaml

Be aware that the `latest` tag is not a stable release.

**Reply by heidemn** [reply]
Dex v2.27.0 has limited support for env variable interpolation in the config.
It's only enabled for connectors and storage.

In the following example, the value from env variable `MY_POSTGRES_HOST` is used to configure the Postgres host.

```yaml
storage:
  type: postgres
  config:
    host: '$MY_POSTGRES_HOST'
    ...
```

This env expansion is entirely text-based and not aware of YAML syntax. It's done before the YAML parsing.
If your env variable e.g. contains single quotes, you would have to escape them.

Compared to that, the new `gomplate` way in the next Dex version is definitely nicer and more powerful.
It allows to use env variables anywhere, and brings in even more functionality to customize the Dex config.

**Reply by AnkitaBarbi** [reply]
but cant use latest image as its not stable , hence was trying to look into 2.27.0. 
I did a work around to make it work - where created .yaml file using EOF in shell script and Entrypoint in Dockerfiles is pointed to shell script.
